src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/PackageSourceTelemetry.cs

```diff
@@ -209,13 +213,64 @@ public async Task SendTelemetryAsync()
                     sourceRepository = new SourceRepository(new PackageSource(source), Repository.Provider.GetCoreV3());
                 }
 
+                Uri sourceUri = null;
+                bool hasAnHttpResource = false;
+
+                try
+                {
+                    sourceUri = new Uri(source);
+                }
+                catch (ArgumentException)
+                {
+
+                }
+                catch (UriFormatException)
+                {
+
+                }
+                finally
+                {
+                    if (sourceUri != null)
+                    {
+                        if (sourceUri.Scheme == Uri.UriSchemeHttps)
+                        {
+                            NumberOfHTTPSSources++;
+                        }
+                    }
+                }
+
+
+                if (sourceRepository.GetServiceIndexV3FromCache(out ServiceIndexResourceV3 resource))
```

Comment:
Would the protocol diagnostics event be more appropriate instead of going through this method?

@zivkan wrote up a documented on that at some point but I can't find it now.

Reply:
Hoping @zivkan can find the doc for this discussion.
Wondering if we can use ProtocolDiagnostics instead. 

The idea is that something happens, data is reported, and we consume it use at a different level.

Reply:
I can't find a design for ProtocolDiagnostics itself, but I can find this doc on different "monitoring" APIs available to NuGet: https://github.com/NuGet/Client.Engineering/blob/main/designs/ProductMonitoring.md

The overview of how ProtocolDiagnostics works is that NuGet.Protocol have some events that get raised when potentially "interesting" things happen (http requests are made, or one of `SourceRepository`'s "resources" are used). Then the code in VS can subscribe to those events and react to them appropriately.  They way I implemented it is subscribe to the event at the start of restore, and each time the event fires, increment a counter in memory. Once restore finishes, ubsubscribe from the event, and emit telemetry about the counts that were accumulated.

Nikolche's idea is a good one. It does seem appropriate that we could report this stuff in the same protocol diagnostics event we're already emiting. Although adding it to one of these events is also fine IMO. Just as long as it's new properties on an existing event, and not a new telemetry event.

Reply:
I used a similar pattern here https://github.com/NuGet/NuGet.Client/pull/6125